pdc_client/bin/pdc

```diff
@@ -0,0 +1,321 @@
+#! /usr/bin/env python
+
+import json
+import sys
+import argparse
+import beanbag
+import requests
+import requests_kerberos
+import os
+import os.path
+import warnings
+import itertools
+import logging
+import imp
+import argcomplete
+
+
+PLUGIN_DIRS = [
+    os.path.join(os.path.dirname(__file__), '..', 'plugins')
```

Comment:
Supposing the client should be open-source, we need some way to extend it to work with internal parts. This was the easiest way to achieve that. The client finds all python modules in a predefined place, loads the modules and later calls functions in them (`run_hook`). The internal things would just implement the module, drop it into proper place and the client would just work.